neuralnetwork/machinelearning/NeuralNetwork.java

```diff
@@ -1,76 +1,87 @@
+        return new NeuralNetwork(learningRate);
+    }
+
+    public NeuralNetwork addInputLayer(int size) throws Exception {
         if (!(inputLayer == null)) {
             throw new Exception("Error : Input layer already exists!");
         }
-        this.inputLayer = new InputLayer(inputFunction, size);
+        this.inputLayer = new InputLayer(inputNormalize, size);
         this.inputLayerSize = size;
+        return this;
     }
 
-    public void addHiddenLayer(ActivateFunction function, int size) throws Exception {
+    public NeuralNetwork addHiddenLayer(int size) throws Exception {
         if (inputLayer == null) {
             throw new Exception("Error : Input layer must be added before adding hidden layer");
         }
         if (!(outputLayer == null)) {
             throw new Exception("Error : Can't add hidden layer when output layer is already added");
         }
         if (hiddenLayers.isEmpty()) {
-            hiddenLayers.add(new HiddenLayer(function, inputLayerSize, size, this.learningRate));
+            hiddenLayers.add(new HiddenLayer(activateFunction, inputLayerSize, size, this.learningRate));
             layerSize.add(new int[]{inputLayerSize, size});
         } else {
             int temp = hiddenLayersSize.get(hiddenLayers.size() - 1);
-            hiddenLayers.add(new HiddenLayer(function, temp, size, this.learningRate));
+            hiddenLayers.add(new HiddenLayer(activateFunction, temp, size, this.learningRate));
             layerSize.add(new int[]{temp, size});
         }
         hiddenLayersSize.add(size);
+        return this;
     }
 
-    public void addOutputLayer(ActivateFunction function, int size) throws Exception {
+    public NeuralNetwork addOutputLayer(int size) throws Exception {
         if (inputLayer == null) {
             throw new Exception("Error : Input layer must be added before adding input layer");
         }
         if (!(outputLayer == null)) {
             throw new Exception("Error : Output layer already exists!");
         }
         if (hiddenLayers.isEmpty()) {
-            outputLayer = new OutputLayer(function, inputLayerSize, size, this.learningRate);
+            outputLayer = new OutputLayer(softmax, inputLayerSize, size, this.learningRate);
             layerSize.add(new int[]{inputLayerSize, size});
         } else {
             int temp = hiddenLayersSize.get(hiddenLayers.size() - 1);
-            outputLayer = new OutputLayer(function, temp, size, this.learningRate);
+            outputLayer = new OutputLayer(softmax, temp, size, this.learningRate);
             layerSize.add(new int[]{temp, size});
         }
+        return this;
     }
 
     public ArrayList<LayerData> getLayerData() throws Exception {
-        if (outputLayer == null) {
-            throw new Exception("Error : output layer must be added before saving settings");
-        }
         ArrayList<LayerData> layerDataArrayList = new ArrayList<>();
```

Comment:
This method is called for saving weight and bias, so i think it should check whether output layer has added